cpp/include/cudf/detail/valid_if.cuh

```diff
@@ -48,7 +48,7 @@ __global__ void valid_if_kernel(
 {
   constexpr size_type leader_lane{0};
   auto const lane_id{threadIdx.x % warp_size};
-  size_type i = threadIdx.x + blockIdx.x * blockDim.x;
+  std::size_t i = threadIdx.x + blockIdx.x * blockDim.x;
```

Comment:
Don't you get a comparison of signed and unsigned error/warning on line 55 now?

Reply:
Interestingly, no.

Reply:
I'm not sure if by this declaration, the numbers on the right side are converted to size_t before multiplication, or they are still int32_t and only the final result is converte.

Reply:
Even it is the right/correct implementation (i.e. the numbers are converted before `+` and `*`), this is confusing and error-prone. I would suggest using some other way that explicit. "grid-stride range helper" as mentioned below can be a solution.

Reply:
Then I think `int64_t` would be a better choice here, because it can both eliminate overflow and avoid signed/unsigned warning issues.

Reply:
Added the thread_index_type.